essim/src/ssim_ssim.c

```diff
@@ -298,6 +437,28 @@ ssim_allocate_ctx_array(const size_t numCtx, const uint32_t width,
   memset(p->ctx, 0, sizeof(arraySize));
   p->numCtx = numCtx;
 
+#if UPDATED_INTEGER_IMPLEMENTATION
+  uint32_t SSIMValRtShiftBits = 0;
+  uint32_t SSIMValRtShiftHalfRound = 0;
+  uint32_t* div_lookup_ptr = NULL;
+  const uint64_t MAX_SSIM_ACCUMULATED_SUM_VALUE = (uint64_t)1 << 63;
+  if(mode != SSIM_MODE_PERF_FLOAT) {
+    /*generating LUT to avoid final stage division in cal window for ssim_val*/
+    div_lookup_ptr = div_lookup_generator();
+
+    if(essim_mink_value == 4) {
+      const uint64_t MAX_SSIM_ACCUMULATED_SUM_VALUE = 18446744073709551615;
```

Comment:
this also gives a 

```
unused variable 'MAX_SSIM_ACCUMULATED_SUM_VALUE' [-Wunused-variable]
```

which I'm pretty sure means this isn't working as intended, rather than overriding `MAX_SSIM_ACCUMULATED_SUM_VALUE` we are simply shadowing it inside of the if block. This should probably be made into not a const so it can be overriden appropriately or just check the condition at the time it is declared like

```
const uint64_t MAX_SSIM_ACCUMULATED_SUM_VALUE = essim_mink_value == 4 ? 18446744073709551615U : (uint64_t)1 << 63;
```